src/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,25 @@
+import {Image} from 'react-native-compressor';
+import {composeImageMeta} from '../utils/string/file';
+import {uploadPhoto} from '../service/file';
+
+const useUploadImage = () => {
+  const uploadPhotoImage = async (pathImg: string) => {
+    try {
+      const compressionResult = await Image.compress(pathImg, {
+        compressionMethod: 'auto'
+      });
+      const asset = new FormData();
+      asset.append('photo', composeImageMeta(compressionResult));
+      const responseUpload = await uploadPhoto(asset);
+      return responseUpload;
+    } catch (e) {
+      return undefined;
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `uploadPhotoImage` function is swallowing any exceptions that occur during the image compression or upload process. This could make it difficult to debug issues since no error information will be available. Instead of returning `undefined`, consider rethrowing the error after optionally logging it.

```diff
    } catch (e) {
      console.error(e);
-      return undefined;
+      throw e;
    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->